app/src/main/java/com/everest/moviedb/viewmodel/MovieViewModel.kt

```diff
@@ -0,0 +1,56 @@
+package com.everest.moviedb.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.everest.moviedb.models.Movie
+import com.everest.moviedb.network.MovieRepository
+import kotlinx.coroutines.launch
+
+sealed class MovieData {
+    class Data(val movieList: List<Movie>) : MovieData()
+    class Error(val errorMessage: String?) : MovieData()
+}
+
+class MovieViewModel(private val movieRepository: MovieRepository) : ViewModel() {
+
+    private val _movieData = MutableLiveData<MovieData>()
```

Comment:
I can see that you are posting data and error in same type and UI has to check the type.
Instead of that can you please post it as single State or individually also fine.